File: 1046 - Last Stone Weight.java
import java.util.PriorityQueue;

class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> q = new PriorityQueue<>((a, b) -> b - a);
        for (int s : stones) {
            q.offer(s);
        }
        while (q.size() > 1) {
            int a = q.poll();
            int b = q.poll();
            if (a > b) {
                q.offer(a - b);
            }
        }
        int res;
        if (q.size() == 0) res = 0;
        else res = q.poll();
        return res;
    }
}

File: 1046 - Last Stone Weight.cpp
#include <queue>

using namespace std;

class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> q;
        for (int s : stones) q.push(s);

        while (q.size() > 1) {
            int a = q.top();
            q.pop();
            int b = q.top();
            q.pop();
            if (a > b) {
                q.push(a - b);
            }
        }
        int res;
        if (q.size() == 0) res = 0;
        else res = q.top();
        return res;
    }
};